### PR TITLE
full text search

### PR DESCRIPTION
**Pull request by yrong**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [y] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [n] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [y] Have you added new tests to prevent regressions?
- [y] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [y] Did you follow the commit message conventions explained in CONTRIBUTING.md?

### CODE REVIEW

File: lib/operators.js [hunk 1]
```diff
@@ -80,7 +81,8 @@ const Aliases = {
   $all: Op.all,
   $values: Op.values,
   $col: Op.col,
-  $raw: Op.raw  //deprecated remove by v5.0
+  $raw: Op.raw,  //deprecated remove by v5.0
+  $fulltext:Op.fulltext
```
Comment (sushantdhiman) [inline lib/operators.js]: space after `:`

File: lib/dialects/postgres/query-generator.js [hunk 2]
```diff
@@ -896,6 +896,10 @@ const QueryGenerator = {
     }
 
     return AbstractQueryGenerator.setAutocommitQuery.call(this, value, options);
+  },
+
+  fullTextSearchQuery(value, options) {
+    return `plainto_tsquery(${this.escape(options.regconfig||'english')}, ${this.escape(value)})`;
```
Comment (sushantdhiman) [inline lib/dialects/postgres/query-generator.js]: spaces before/after `||`, I really need to add these to eslint rules :D

File: test/unit/dialects/postgres/query-generator.test.js [hunk 3]
```diff
@@ -532,6 +532,13 @@ if (dialect.match(/^postgres/)) {
           arguments: ['myTable', {where: {field: {$notIRegexp: '^[h|a|t]'}}}],
           expectation: "SELECT * FROM \"myTable\" WHERE \"myTable\".\"field\" !~* '^[h|a|t]';",
           context: QueryGenerator
+        },
+        {
+          title: 'generate fulltext search',
+          arguments: ['myTable', {where: {field: {$ts: 'foo'}}}],
+          expectation: 'SELECT * FROM myTable WHERE myTable.field @@ plainto_tsquery(\'english\', \'foo\');',
+          context: {options: {quoteIdentifiers: false, regconfig:'english'}},
```
Comment (mkaufmaner) [inline test/unit/dialects/postgres/query-generator.test.js]: There should also be a test where `quoteIdentifiers: true`

Comment (mkaufmaner) [reply test/unit/dialects/postgres/query-generator.test.js]: This is also related to https://github.com/sequelize/sequelize/pull/8404

File: lib/dialects/abstract/query-generator.js [hunk 4]
```diff
@@ -1985,7 +1985,8 @@ const QueryGenerator = {
     [Op.or]: ' OR ',
     [Op.col]: 'COL',
     [Op.placeholder]: '$$PLACEHOLDER$$',
-    [Op.raw]: 'DEPRECATED' //kept here since we still throw an explicit error if operator being used remove by v5,
+    [Op.raw]: 'DEPRECATED', //kept here since we still throw an explicit error if operator being used remove by v5,
+    [Op.ts]: '@@',
```
Comment (eseliger) [inline lib/dialects/abstract/query-generator.js]: I would keep raw the last element, as its deprecated

File: lib/operators.js [hunk 5]
```diff
@@ -80,7 +81,8 @@ const Aliases = {
   $all: Op.all,
   $values: Op.values,
   $col: Op.col,
-  $raw: Op.raw  //deprecated remove by v5.0
+  $raw: Op.raw,  //deprecated remove by v5.0
+  $ts: Op.ts
```
Comment (eseliger) [inline lib/operators.js]: same here

**Comment by samanmohamadi** [comment]
any news ?

File: lib/dialects/abstract/query-generator.js [hunk 6]
```diff
@@ -2335,6 +2336,8 @@ const QueryGenerator = {
       case Op.between:
       case Op.notBetween:
         return this._joinKeyValue(key, `${this.escape(value[0])} AND ${this.escape(value[1])}`, comparator, options.prefix);
+      case Op.ts:
```
Comment (jonykalavera) [inline lib/dialects/abstract/query-generator.js]: Is this necessary? I think this really limits the use cases of this operator. https://www.postgresql.org/docs/current/static/functions-textsearch.html 

Maybe the same could be achieved using `sequelize.fn`:

```javascript
Post.findAll({
  where: {
    my_ts_vector: {
      [Op.ts]: sequielize.fn('plainto_tsquery', 'english', 'my ts query')
    }
  }
});
```

What do you think?

Comment (jonykalavera) [reply lib/dialects/abstract/query-generator.js]: @yrong Forked your fork (LOL). https://github.com/jonykalavera/sequelize/commit/bd2da2b23291df50333a5b8660c822777b3951cd

This allows me to build the ts query myself. Do you think we can get to a middle ground?

**Comment by jordanell** [comment]
Would love to see this PR merged or one like it as there appears to have been numerous attempts such as https://github.com/sequelize/sequelize/pull/8404 and https://github.com/sequelize/sequelize/pull/8302.

Is there any movement here? If not I would definitely get involved to get one of these merged.

**Comment by jordanell** [comment]
@mkaufmaner are you open for a PR related to this functionality against the version 5 branch at all?

**Comment by sushantdhiman** [comment]
Cleanup, closing this PR. If someone wants to takeover please open a new PR

**Pull request closed without merging**